test/dist_autograd_test.py

```diff
@@ -160,3 +198,257 @@ def test_rpc_complex_args(self):
                     self.assertEqual(tensors[i], next_funcs[i][0].variable)
                 else:
                     self.assertIsNone(next_funcs[i][0])
+
+    @dist_init
+    def test_error_in_context(self):
+        with dist_autograd.context() as context_id:
+            t1 = torch.rand(3, 3, requires_grad=True)
+            t2 = torch.rand(6, 6, requires_grad=True)
+
+
+            with self.assertRaises(RuntimeError):
+                # This should throw an error since matrix sizes don't match.
+                dist.rpc_sync('worker{}'.format(self._next_rank()), torch.matmul,
+                              args=(t1, t2))
+
+    def _verify_backwards(self, exec_mode, tensors, context_id, local_grads, *args):
+        if exec_mode == ExecMode.REMOTE:
+            self._verify_backwards_remote(tensors, context_id, local_grads, *args)
+        else:
+            torch.autograd.backward(tensors)
+            return [arg.grad for arg in args]
+
+    def _verify_backwards_remote(self, tensors, context_id, local_grads, *args):
```

Comment:
Q: is any part of this is running on a remote worker? why this is `remote` in the name? Or does `get_gradient` returns all gradient including remote ones?

Reply:
The `remote` mostly refers to `REMOTE` execution mode and the fact that we're using distributed autograd instead of the local autograd engine.